src/components/form/VersesField.tsx

```diff
@@ -0,0 +1,259 @@
+    },
+    isEqual: compareNullable((a, b) => areListsDeepEqual(a, b)),
+    ...props,
+  });
+  const disabled = disabledProp ?? meta.submitting;
+  const ref = useFocusOnEnabled(meta, disabled);
+  const [scriptureRanges, setScriptureRanges] = useState<Val>(
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    input.value ?? []
+  );
+
+  const [_, end] = inputValue.split('-');
+
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  const [__, endVerse] = end?.split(':') ?? [undefined, undefined];
+
+  const hideVerseOrderError =
+    errorCode === 'verseOrder' &&
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    (!end || !end.includes(':') || !endVerse || endVerse?.length < 1);
+
+  const hideChapterOrderError =
+    errorCode === 'chapterOrder' && (!end || !end.includes(':'));
+
+  const hideVerseNumberError =
+    errorCode === 'invalidVerse' &&
+    end &&
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    (!end.includes(':') || endVerse?.length < 1);
+
+  const error =
+    showError(meta) ||
+    (meta.error &&
+      !hideVerseOrderError &&
+      !hideChapterOrderError &&
+      !hideVerseNumberError &&
+      meta.error !== 'Required');
+
+  const helperText = getHelperText(
+    meta,
+    error ? meta.error : helperTextProp || ' ',
+    error
+  );
+  return (
+    <Autocomplete<GenericScriptureRange | string, true, false, true>
+      {...rest}
+      classes={props.classes}
+      multiple
+      options={[]}
+      freeSolo
+      renderTags={(value, getTagProps) => {
+        return value.map((option, index) => {
+          if (!option) return null;
+          if (typeof option !== 'string') option = formatScriptureRange(option);
+          return (
+            <Chip
+              variant="outlined"
+              {...ChipProps}
+              label={option}
+              {...getTagProps({ index })}
+              classes={{
+                // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+                ...((ChipProps as ChipProps)?.classes ?? {}),
+                deleteIcon: classes.chipDeleteIcon,
+              }}
+              className={classes.chip}
+            />
+          );
+        });
+      }}
+      value={scriptureRanges}
+      inputValue={inputValue}
+      onInputChange={(_, value) => {
+        if (errorCode && !meta.error) setErrorCode('');
+        const cleaned = value.replace(/[^\d:-]/g, '');
+        const validated = cleaned ? validateInput(cleaned) : '';
+        if (validated === null) return;
+        input.onChange(validated);
+        setInputValue(validated);
+      }}
+      onChange={(_, value) => {
+        if (meta.error) {
+          setErrorCode('createError');
+          // it'd be nice to be able to preserve the input value if a user tries submitting when an error is present
+          // setInputValue(value[value.length - 1] as string);
```

Comment:
The bug I mentioned is that if there's an error and the user hits enter, the input is erased. I was manually setting the input value before, but this caused other problems with FF's state. This code is planned to be eventually refactored anyhow.